syne_tune/optimizer/schedulers/transfer_learning/rush.py

```diff
@@ -0,0 +1,151 @@
+            transfer_learning_evaluations=transfer_learning_evaluations,
+            metric=metric,
+            mode=kwargs.get('mode', 'min')
+        )
+        if 'points_to_evaluate' in kwargs:
+            points_to_evaluate += kwargs['points_to_evaluate']
+            points_to_evaluate = [dict(s) for s in set(frozenset(p.items()) for p in points_to_evaluate)]
+        kwargs['points_to_evaluate'] = points_to_evaluate
+
+        self._hyperband_scheduler = HyperbandScheduler(config_space, metric=metric, **kwargs)
+        self._num_init_configs = len(points_to_evaluate)
+        self._thresholds = dict()  # thresholds at different resource levels that must be met
+
+    @staticmethod
+    def _determine_baseline_configurations(config_space: Dict,
+                                           transfer_learning_evaluations: Dict[str, TransferLearningTaskEvaluations],
+                                           metric: str,
+                                           mode: str) -> List[Dict]:
+        argbest, best = (np.argmin, np.min) if mode == 'min' else (np.argmax, np.max)
+        baseline_configurations = list()
+        for evals in transfer_learning_evaluations.values():
+            best_hpc_idx = argbest(best(evals.objective_values(objective_name=metric)[..., -1], axis=1))
+            hpc = evals.hyperparameters.iloc[best_hpc_idx]
+            baseline_configurations.append({
+                key: hpc[key] for key in config_space
+            })
+        return baseline_configurations
+
+    def on_trial_error(self, trial: Trial) -> None:
+        self._hyperband_scheduler.on_trial_error(trial)
+
+    def on_trial_result(self, trial: Trial, result: Dict) -> str:
+        trial_decision = self._hyperband_scheduler.on_trial_result(trial, result)
+        return self._on_trial_result(trial_decision, trial, result)
+
+    def on_trial_remove(self, trial: Trial) -> None:
+        self._hyperband_scheduler.on_trial_remove(trial)
+
+    def on_trial_complete(self, trial: Trial, result: Dict) -> None:
+        self._hyperband_scheduler.on_trial_complete(trial, result)
+
+    def suggest(self, trial_id: int) -> Optional[TrialSuggestion]:
+        return self._hyperband_scheduler.suggest(trial_id)
+
+    def state_dict(self) -> Dict:
+        record = self._hyperband_scheduler.state_dict()
+        record['num_init_configs'] = pickle.dumps(self._num_init_configs)
+        record['thresholds'] = pickle.dumps(self._thresholds)
+        return record
+
+    def load_state_dict(self, state_dict):
+        self._num_init_configs = pickle.loads(state_dict['num_init_configs'])
+        del state_dict['num_init_configs']
+        self._thresholds = pickle.loads(state_dict['thresholds'])
+        del state_dict['thresholds']
+        self._hyperband_scheduler(state_dict)
+
+    def _on_trial_result(self, trial_decision: str, trial: Trial, result: Dict) -> str:
+        if trial_decision != SchedulerDecision.CONTINUE:
```

Comment:
Comment above. For type == 'promotion', you will never see CONTINUE, so your modification would not do anything. The promotion type is implement in suggest.

It would, I think, be simpler to implement another RungSystem